sound/soc/sof/intel/hda-dsp.c

```diff
@@ -412,8 +412,7 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 		 * But it can be kept enabled when the DSP enters D0I3 while the
 		 * system is in S0 for debug.
 		 */
-		if (hda_enable_trace_D0I3_S0 &&
-		    sdev->system_suspend_target != SOF_SUSPEND_NONE)
+		if (sdev->system_suspend_target != SOF_SUSPEND_NONE)
```

Comment:
NAK. it's a kernel parameter added on purpose.
module_param_named(enable_trace_D0I3_S0, hda_enable_trace_D0I3_S0, bool, 0444);
MODULE_PARM_DESC(enable_trace_D0I3_S0,
		 "SOF HDA enable trace when the DSP is in D0I3 in S0"

why do you need to remove this?

Reply:
@plbossart it is causing premature WoV wakeup

Reply:
@mrajwa I think we should just change it to if (!hda_enable_trace_D0I3_S0 ||
		    sdev->system_suspend_target != SOF_SUSPEND_NONE)

Reply:
Agree there's likely a logical confusion here, but if the traces were enabled I am not sure how we ever entered S0ix on other platforms. Something else must have changed?

Reply:
In my DUT, I see this:
[   44.128231] hda_dsp_set_D0_state 415 hda_enable_trace_D0I3_S0=0

So, how will this affect trace issue?

Reply:
> In my DUT, I see this:
> [ 44.128231] hda_dsp_set_D0_state 415 hda_enable_trace_D0I3_S0=0
> 
> So, how will this affect trace issue?

Ignore my question please, I see we are setting HDA_PM_NO_DMA_TRACE flag.. got it.

Reply:
> Agree there's likely a logical confusion here, but if the traces were enabled I am not sure how we ever entered S0ix on other platforms. Something else must have changed?

@plbossart previous platforms had a bit more leniency in terms of S0ix entry qualifications. But with TGL, we dont have that anymore. But yes this logic was flawed here.

Reply:
@ranj063 @plbossart yes, the logic here is incorrect, it should be something like this:
```
		/*
		 * Trace DMA is disabled by default when the DSP enters D0I3.
		 * But it can be kept enabled when the DSP enters D0I3 while the
		 * system is in S0 (SUSPEND_NONE) for debug.
		 */
		if (!(hda_enable_trace_D0I3_S0 &&
		      sdev->system_suspend_target == SOF_SUSPEND_NONE))
			flags = HDA_PM_NO_DMA_TRACE;
```